mllib/src/main/scala/org/apache/spark/mllib/classification/NaiveBayes.scala

```diff
@@ -326,6 +326,8 @@ class NaiveBayes private (
   /** Set the smoothing parameter. Default: 1.0. */
   @Since("0.9.0")
   def setLambda(lambda: Double): NaiveBayes = {
+    require(lambda > 0,
+      s"Smoothing parameter must be greater than 0 but got ${lambda}")
```

Comment:
Yeah this is looking good, to get all of the same obvious parameter checks done in one pass. You can say "positive" for "greater than 0" and "nonnegative" for "greater than or equal to 0" but this is fine.